src/Classes/ExchangeRate.php

```diff
@@ -1,377 +1,28 @@
 <?php
 
+declare(strict_types=1);
+
 namespace AshAllenDesign\LaravelExchangeRates\Classes;
 
-use AshAllenDesign\LaravelExchangeRates\Exceptions\ExchangeRateException;
-use AshAllenDesign\LaravelExchangeRates\Exceptions\InvalidCurrencyException;
-use AshAllenDesign\LaravelExchangeRates\Exceptions\InvalidDateException;
-use Carbon\Carbon;
-use Exception;
-use GuzzleHttp\Client;
-use GuzzleHttp\Exception\GuzzleException;
+use AshAllenDesign\LaravelExchangeRates\Drivers\ExchangeRatesApiIo\ExchangeRatesApiIoDriver;
+use AshAllenDesign\LaravelExchangeRates\Drivers\ExchangeRatesDataApi\ExchangeRatesDataApiDriver;
+use AshAllenDesign\LaravelExchangeRates\Interfaces\ExchangeRateDriver;
+use Illuminate\Support\Manager;
 
-class ExchangeRate
+class ExchangeRate extends Manager
 {
-    /**
-     * The object used for making requests to the currency
-     * conversion API.
-     *
-     * @var RequestBuilder
-     */
-    private $requestBuilder;
-
-    /**
-     * The repository used for accessing the cache.
-     *
-     * @var CacheRepository
-     */
-    private $cacheRepository;
-
-    /**
-     * Whether of not the exchange rate should be cached
-     * after being fetched from the API.
-     *
-     * @var bool
-     */
-    private $shouldCache = true;
-
-    /**
-     * Whether or not the cache should be busted and a new
-     * value should be fetched from the API.
-     *
-     * @var bool
-     */
-    private $shouldBustCache = false;
-
-    /**
-     * ExchangeRate constructor.
-     *
-     * @param  RequestBuilder|null  $requestBuilder
-     * @param  CacheRepository|null  $cacheRepository
-     */
-    public function __construct(RequestBuilder $requestBuilder = null, CacheRepository $cacheRepository = null)
+    public function createExchangeRatesDataApiDriver(): ExchangeRateDriver
     {
-        $this->requestBuilder = $requestBuilder ?? new RequestBuilder(new Client());
-        $this->cacheRepository = $cacheRepository ?? new CacheRepository();
+        return new ExchangeRatesDataApiDriver();
```

Comment:
I'm not sure about this but, wouldn't be more extensible to move this into the service provider and the binding the different drivers instead having this in the class? Let's say:

```php
$manager = app('exchange-rate');

$manager->extend('data-api-driver', fn($app) => new ExchangeRatesDataApiDriver());
```

So, in the case that the app requires new drivers doesn't have to be coupled to the class by inheritance. Let me know what you think.

Reply:
I'm not too sure about this either, but I could definitely see a use case for it. I've just had a quick look at how Laravel's using this approach, and it looks like they do it the same as me: https://github.com/laravel/framework/blob/9.x/src/Illuminate/Hashing/HashManager.php

So I think I'm cool with how it is at the moment, but I'm definitely open to changing it in the future though 🙂